File: assets/src/pet.cpp
// Pet sample

#include <iostream>
#include <string>

using namespace std;

class Pet
{
public:
    virtual std::string GetAngry() = 0;
    virtual ~Pet() {}
};

class Cat : public Pet
{
public:
    virtual std::string GetAngry()
    {
        return "ネコがひっかく";
    }
};


class Dog : public Pet
{
public:
    virtual std::string GetAngry()
    {
        return "イヌがほえる";
    }
};

int main()
{
	Pet* pet = new Cat();
    std::cout << pet->GetAngry() << std::endl;
    delete pet;

	pet = new Dog();
    std::cout << pet->GetAngry() << std::endl;
    delete pet;

    return 0;
}

File: assets/src/pet.rb
# Pet sample

class Pet
   def getAngry
   end
end

class Cat < Pet
   def getAngry
      "ネコがひっかく"
   end
end

class Dog < Pet
   def getAngry
      "イヌがほえる"
   end
end

pet = Cat.new
puts pet.getAngry
pet = Dog.new
puts pet.getAngry
